src/NHSD.BuyingCatalogue.Ordering.Persistence/Repositories/OrderRepository.cs

```diff
@@ -1,76 +1,76 @@
-            }
-
-            using ( var dbContextTransaction = _context.Database.BeginTransaction())
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using static System.Int32;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using NHSD.BuyingCatalogue.Ordering.Application.Persistence;
+using NHSD.BuyingCatalogue.Ordering.Domain;
+using NHSD.BuyingCatalogue.Ordering.Persistence.Data;
+
+namespace NHSD.BuyingCatalogue.Ordering.Persistence.Repositories
+{
+    public sealed class OrderRepository : IOrderRepository
+    {
+        public const string DefaultOrderId = "C000000-01";
+        private readonly ApplicationDbContext _context;
+
+        public OrderRepository(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IEnumerable<Order>> ListOrdersByOrganisationIdAsync(Guid organisationId)
+        {
+            return await _context.Order.Include(x => x.OrderStatus).Where(o => o.OrganisationId == organisationId)
+                .ToListAsync();
+        }
+
+        public async Task<Order> GetOrderByIdAsync(string orderId)
+        {
+            return await _context.Order.FindAsync(orderId);
+        }
+
+        public async Task UpdateOrderAsync(Order order)
+        {
+            if (order is null)
+            {
+                throw new ArgumentNullException(nameof(order));
+            }
+
+            _context.Order.Update(order);
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task<string> GetLatestOrderIdByCreationDate()
+        {
+            var latestOrder = await _context.Order.OrderByDescending(o => o.Created).FirstOrDefaultAsync();
+            return latestOrder?.OrderId;
+        }
+
+        public async Task<string> CreateOrderAsync(Order order)
+        {
+            if (order is null)
+            {
+                throw new ArgumentNullException(nameof(order));
+            }
+
+            using (var dbContextTransaction = _context.Database.BeginTransaction())
             {
                 if (order.OrderId == null)
                 {
                     order.OrderId = await GetIncremenedOrderId();
                 }
 
-                order.OrderStatus = await  _context.OrderStatus.FindAsync(order.OrderStatus.OrderStatusId) ?? order.OrderStatus;
+                order.OrderStatus = await _context.OrderStatus.FindAsync(order.OrderStatus.OrderStatusId) ?? order.OrderStatus;
 
                 _context.Order.Add(order);
-                await _context.SaveChangesAsync();
-                await dbContextTransaction.CommitAsync();
-            }
-
-            return order.OrderId;
+                await _context.SaveChangesAsync();
+                await dbContextTransaction.CommitAsync();
+            }
+
+            return order.OrderId;
         }
 
         private async Task<string> GetIncremenedOrderId()

```

Comment:
While you're here would you mind tidying up this method?

The method name should be `GetIncrementedOrderId`.

Could you also tidy up the call to `int.Parse` by removing the redundant argument name (`s:`) and simplify `System.Globalization.CultureInfo.InvariantCulture` by introducing a `using`?

I'd also be tempted to remove the `using static System.Int32;` and include the type name instead, e.g. `int.Parse`.

Reply:
👍